tools/golang/cmd/encrypt/script.go

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2019 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package main
+
+import (
+	"errors"
+	"flag"
+	tinkpb "github.com/google/tink/proto/tink_go_proto"
+	"github.com/subscriptions-project/swg-js/tools/golang/encryption"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+)
+
+/* Script to encrypt documents for the SwG Encryption Project.
+*
+* Golang script which takes in an input HTML document and encrypts
+* all content within <section subscriptions-section="content" encrypted>
+* tags using AES-GCM. The key used to encrypt the content is added
+* to the output document's head inside of a
+* <script cryptokeys type="application/json"> element. The encrypted
+* document is outputted to the output_file path given as a flag.
+*
+* Example Usage:
+* go run swg-js/tools/golang/cmd/encrypt/script.go \
+*	--input_html_file=../tmp/sample-encryption.html \
+*	--output_file=../tmp/sample-encryption-out.html \
+*	--google_public_key_url=https://news.google.com/swg/encryption/keys/{dev|prod}/tink/public_key \
+*	--access_requirement=norcal.com:premium \
+*	--publisher_public_key_url=nytimes.com,www.nytimes.com/scs/publickey \
+*	--publisher_public_key_url=wp.com,www.wp.com/scs/publickey
+ */
+type mapFlags map[string]string
+
+const googleDevPublicKeyUrl string = "https://news.google.com/swg/encryption/keys/dev/tink/public_key"
+
+func (m *mapFlags) String() string {
+	var strs []string
+	for key, val := range *m {
+		strs = append(strs, key, ";", val)
```

Comment:
I'll use a comma in the String method here. I would like to avoid using the equals sign because I think that it can be confusing to have an equals sign for the flag and within the value itself. I can't use semi-colons in the flag because that is a command separator for the command line.